### PR TITLE
Implement debugger variables view

### PR DESCRIPTION
**Pull request by kenzieschmoll**
Screen Shot 2020-05-04 at 3 17 32 PM

### CODE REVIEW

File: packages/devtools_app/lib/src/debugger/flutter/common.dart [hunk 1]
```diff
@@ -45,3 +45,18 @@ Widget createAnimatedCircleWidget(double radius, Color color) {
     decoration: BoxDecoration(color: color, shape: BoxShape.circle),
   );
 }
+
+TextStyle regularStyle(BuildContext context) {
```
Comment (jacob314) [inline packages/devtools_app/lib/src/debugger/flutter/common.dart:49]: consider making these extension methods on the context.

File: packages/devtools_app/lib/src/debugger/flutter/debugger_controller.dart [hunk 2]
```diff
@@ -462,12 +466,97 @@ class DebuggerController extends DisposableController
   void selectStackFrame(StackFrameAndSourcePosition frame) {
     _selectedStackFrame.value = frame;
 
+    if (frame?.frame != null) {
```
Comment (devoncarew) [inline packages/devtools_app/lib/src/debugger/flutter/debugger_controller.dart:469]: Is is possible for the `frame` field on a `StackFrameAndSourcePosition` instance to be null?

File: packages/devtools_app/lib/src/debugger/flutter/debugger_controller.dart [hunk 3]
```diff
@@ -462,12 +466,97 @@ class DebuggerController extends DisposableController
...
+      _variables.value = [];
+    }
+
     if (frame?.scriptRef != null) {
       showScriptLocation(
           ScriptLocation(frame.scriptRef, location: frame.position));
     }
   }
 
+  List<Variable> _variablesForFrame(Frame frame) {
```
Comment (devoncarew) [inline packages/devtools_app/lib/src/debugger/flutter/debugger_controller.dart:481]: `_variablesForFrame` ==> `_createVariablesForFrame`?

File: packages/devtools_app/lib/src/debugger/flutter/debugger_controller.dart [hunk 4]
```diff
@@ -462,12 +466,97 @@ class DebuggerController extends DisposableController
...
+    final variables = frame.vars.map((v) => Variable.create(v)).toList();
+    variables.forEach(buildVariablesTree);
+    return variables;
+  }
+
+  Future<void> buildVariablesTree(Variable variable) async {
```
Comment (devoncarew) [inline packages/devtools_app/lib/src/debugger/flutter/debugger_controller.dart:487]: can you add brief docs about what this method does? It doesn't return anything; it populates the children of a variable?

File: packages/devtools_app/lib/src/debugger/flutter/debugger_controller.dart [hunk 5]
```diff
@@ -462,12 +466,97 @@ class DebuggerController extends DisposableController
...
+          variable.addAllChildren(_fieldsAsVariables(result));
+        }
+      }
+    } on SentinelException catch (_) {
+      // Fail gracefully if calling `getObject` throws a SentinelException.
+    }
+    variable.treeInitialized = true;
+  }
+
+  List<Variable> _associationsAsVariables(Instance instance) {
```
Comment (devoncarew) [inline packages/devtools_app/lib/src/debugger/flutter/debugger_controller.dart:508]: _createVariablesForAssociations?

File: packages/devtools_app/lib/src/debugger/flutter/debugger_controller.dart [hunk 6]
```diff
@@ -462,12 +466,97 @@ class DebuggerController extends DisposableController
...
+        .map((BoundField field) {
+          return BoundVariable(
+            name: field.decl.name,
+            value: field.value,
+            scopeStartTokenPos: null,
+            scopeEndTokenPos: null,
+            declarationTokenPos: null,
+          );
+        })
+        .toList()
```
Comment (devoncarew) [inline packages/devtools_app/lib/src/debugger/flutter/debugger_controller.dart:555]: I don't think you need this 1st toList().

File: packages/devtools_app/lib/src/debugger/flutter/common.dart (hunk 1 above)
Comment (devoncarew) [reply packages/devtools_app/lib/src/debugger/flutter/common.dart:49]: Now that these have been moved out of the build methods, perhaps qualify the names slightly more - `regularTestStyle`?

**Comment by jacob314** [comment]
High level feedback on list and map display:
For list display, add a TODO to add custom display for lists with more than 100 elements.
Screen Shot 2020-05-05 at 9 28 19 AM

I think using colons to separate names and values is more consistent with the rest of the devtools UI than using `[` `]`
For example:
```
0: value0
1: value1
2: value2
```

For the map case, add a TODO to make they keys expandable as well as the values any time the type of the Map is not a map of strings, num or another primitive type.

File: packages/devtools_app/lib/src/debugger/flutter/common.dart (hunk 1 above)
Comment (kenzieschmoll) [reply packages/devtools_app/lib/src/debugger/flutter/common.dart:49]: clean! good suggestion

File: packages/devtools_app/lib/src/debugger/flutter/debugger_controller.dart (hunk 2 above)
Comment (kenzieschmoll) [reply packages/devtools_app/lib/src/debugger/flutter/debugger_controller.dart]: good question. changed to `if (frame != null)`

File: packages/devtools_app/lib/src/debugger/flutter/debugger_controller.dart (hunk 3 above)
Comment (kenzieschmoll) [reply packages/devtools_app/lib/src/debugger/flutter/debugger_controller.dart]: done

File: packages/devtools_app/lib/src/debugger/flutter/debugger_controller.dart (hunk 4 above)
Comment (kenzieschmoll) [reply packages/devtools_app/lib/src/debugger/flutter/debugger_controller.dart:492]: done

File: packages/devtools_app/lib/src/debugger/flutter/debugger_controller.dart (hunk 5 above)
Comment (kenzieschmoll) [reply packages/devtools_app/lib/src/debugger/flutter/debugger_controller.dart]: done

File: packages/devtools_app/lib/src/debugger/flutter/debugger_controller.dart (hunk 6 above)
Comment (kenzieschmoll) [reply packages/devtools_app/lib/src/debugger/flutter/debugger_controller.dart]: removed

File: packages/devtools_app/lib/src/debugger/flutter/common.dart [hunk 7]
```diff
@@ -46,17 +46,13 @@ Widget createAnimatedCircleWidget(double radius, Color color) {
   );
 }
 
-TextStyle regularStyle(BuildContext context) {
-  final theme = Theme.of(context);
-  return TextStyle(color: theme.textTheme.bodyText2.color);
-}
+extension DebuggerTextStyleExtension on BuildContext {
```
Comment (devoncarew) [inline packages/devtools_app/lib/src/debugger/flutter/common.dart:49]: Hmm, should these instead be defined on the ThemeData class? I think they're questions we're logically asking of the current theme. See the existing `ThemeDataExtension` extension - `Theme.of(context).isDarkTheme`.

Comment (kenzieschmoll) [reply packages/devtools_app/lib/src/debugger/flutter/common.dart]: done

**Pull request merged**